rclpy/rclpy/parameter_event_handler.py

```diff
@@ -0,0 +1,367 @@
+                                     Otherwise, returns None
+        """
+        if event.node == node_name:
+            for parameter in chain(event.new_parameters, event.changed_parameters):
+                if parameter.name == parameter_name:
+                    return parameter
+
+        return None
+
+    @staticmethod
+    def get_parameters_from_event(event: ParameterEvent) -> Iterable[Parameter]:
+        """
+        Get all parameters from a ParameterEvent message.
+
+        :param event: ParameterEvent message to read
+        """
+        for parameter in chain(event.new_parameters, event.changed_parameters):
+            yield parameter
+
+    def add_parameter_callback(
+        self,
+        parameter_name: str,
+        node_name: str,
+        callback: Callable[[Parameter], None],
+    ) -> ParameterCallbackHandle:
+        """
+        Add new parameter callback.
+
+        Callbacks are called in FILO manner.
+
+        :param parameter_name: Name of a parameter to tie callback to
+        :param node_name: Name of a node, that the parameter should be related to
+        :param callback: A callable to be referenced on a parameter permutation
+
+        :return ParameterCallbackHandle: The duplicate of a callback handle,
+                                         stored in a parameter_event dict
+        """
+        return self._callbacks.add_parameter_callback(
+            parameter_name=parameter_name,
+            node_name=self.resolve_path(node_name),
+            callback=callback,
+        )
+
+    def remove_parameter_callback(
+        self,
+        handle: ParameterCallbackHandle,
+    ) -> None:
+        """
+        Remove a ParameterCallbackHandle.
+
+        Perform no callbacks on this parameter events in the future.
+
+        :param handle: ParameterCallbackHandle of a callback t obe removed
+        """
+        self._callbacks.remove_parameter_callback(handle)
+
+    def add_parameter_event_callback(
+        self,
+        callback: Callable[[ParameterEvent], None],
+    ) -> ParameterEventCallbackHandle:
+        """
+        Add new parameter callback.
+
+        Callbacks are called in FILO manner.
+
+        :param callback: A callable to be referenced on a ParameterEvent
+
+        :return ParameterEventCallbackHandle: The duplicate of a callback handle,
+                                              stored in a parameter_event dict
+        """
+        return self._callbacks.add_parameter_event_callback(callback)
+
+    def remove_parameter_event_callback(
+        self,
+        handle: ParameterEventCallbackHandle,
+    ) -> None:
+        """
+        Remove a ParameterEventCallbackHandle.
+
+        Perform no callbacks on parameter events in the future.
+
+        :param handle: ParameterEventCallbackHandle of a callback t obe removed
+        """
+        self._callbacks.remove_parameter_event_callback(handle)
+
+    def resolve_path(
```

Comment:
Since this is an internal implementation detail of this node, I'd prefer if this was `_resolve_path`.